monkey-runner/plugin/src/test/groovy/com/novoda/monkey/MonkeyRunnerExtensionTest.groovy

```diff
@@ -0,0 +1,51 @@
+package com.novoda.monkey
+
+import org.junit.Before
+import org.junit.Test
+
+import static com.google.common.truth.Truth.assertThat
+
+class MonkeyRunnerExtensionTest {
+
+    private MonkeyRunnerExtension extension
+
+    @Before
+    public void setUp() {
+        extension = new MonkeyRunnerExtension()
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void whenNoTaskDependencySpecified_thenItThrows() {
+        extension.packageNameFilter = 'com.package'
+
+        extension.ensureMandatoryPropertiesPresent()
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void whenNoPackageNameFilterSpecified_thenItThrows() {
+        extension.taskDependency = 'someTask'
+
+        extension.ensureMandatoryPropertiesPresent()
+    }
+
+    @Test
+    public void defaultEventsCountIs50000() {
+        extension.setDefaultsForOptionalProperties()
+
+        assertThat(extension.eventsCount).isEqualTo(50000)
+    }
+
+    @Test
+    public void defaultLogFileNameIsMonkeyLog() {
+        extension.setDefaultsForOptionalProperties()
+
+        assertThat(extension.logFileName).isEqualTo('monkey.log')
+    }
+
+    @Test
+    public void defaultMonkeyTrapIsTrue() {
```

Comment:
I'm not sure they are because atm the defaults are inside the production code, so the tests are as coupled as they can be. IMO I would remove these three.